### PR TITLE
exfat-utils: Replace iconv library by standard C functions mbstowcs() and wcrtomb()

### PR DESCRIPTION
**Pull request by pali**
Via standard C functions mbstowcs() and wcrtomb() is implemented conversion
of strings between exfat's UTF-16LE and multibyte char* in current locale.
There is also standard C macro MB_LEN_MAX which contains maximal length of
one multibyte character.

After this change conversion is not done explicitly to UTF-8, but rather to
encoding this user defined for current locale/codeset. This is also
encoding used by all wide string function in C and also encoding in which
user has configured terminal. Nowadays it is UTF-8, but older systems could
use e.g. Latin1 (or anything other which can be configured by system).
So basically it is encoding which user expects.

iconv library has a problem that cannot be easily used in static linked
binaries or in cross-compiled environments. Moreover iconv() function
has different APIs SUSv2 or POSIX.1-2001/POSIX.1-2008 which differs in
input argument, if is const or not. To somehow workaround this problem,
there is autoconf macro AM_ICONV which defines C macro ICONV_CONST suitable
for type of input argument. But apparently this autoconf macro is broken
for projects which do not use gettext.

As exfat needs only UTF-16LE encoding, it is easier to provide its own
conversion functions implemented via standard C functions, instead of
trying to workaround those iconv problems.

Also code before this commit for handling UTF-8 was incorrect as it
expected that maximal length of UTF-8 sequence is 3 bytes.

### CODE REVIEW

**Comment by namjaejeon** [comment]
Hi Pali,

This patch looks good to me. Could you please check the following ones ?
 1. Add prefix "exfat-utils:" or "others" in patch subject.
 2. Send a pull request to #exfat-next branch, not master.
 3. Run checkpatch.pl in linux kernel source on the patch.

Really thanks for your patch!

**Comment by pali** [comment]
Done!

File: fsck/fsck.c [hunk 1]
```diff
@@ -491,6 +490,14 @@ static bool exfat_boot_region_check(struct exfat *exfat)
...
 }
 
+static size_t utf16_len(const __le16 *str, size_t max_size)
+{
+	size_t i = 0;
+	while (le16_to_cpu(str[i]) && i < max_size)
+		i++;
+	return i;
+}
+
```
[the remark this answers is not in the document]
Comment (pali) [reply fsck/fsck.c:501]: I put it in `fsck.c` because it is not used or needed outside of `fsck.c` file.

[the remark this answers is not in the document]
Comment (hclee) [reply fsck/fsck.c:501]: I know it, But I think it is better to put all of utf related implementation in libexfat.c.

**Pull request merged**